File: generated-code/php/example/TcpReadWrite/Enumeration.php
<?php

namespace  {
    require_once 'Stream.php';

    /**
     * Example enumeration
     */
    abstract class Enumeration
    {
        /**
         * First option
         */
        const VALUE_ONE = 0;

        /**
         * Second option
         */
        const VALUE_TWO = 1;

        /**
         * Read Enumeration from input stream
         */
        public static function readFrom(\InputStream $stream): int
        {
            $result = $stream->readInt32();
            if (0 <= $result && $result < 2) {
                return $result;
            }
            throw new Exception('Unexpected tag value');
        }
    }
}

File: generated-code/pascal/example/TcpReadWrite/Enumeration.pas
unit Enumeration;

{$mode delphi}{$H+}

interface

uses
    Stream,
    SysUtils;

type
    // Example enumeration
    {$scopedEnums on}
    TEnumeration = (
        // First option
        ValueOne = 0,
        // Second option
        ValueTwo = 1);

implementation

end.